src/lib/walletStation.js

```diff
@@ -15,133 +13,224 @@ export const broadcastModeObj = {
-    }
+    redelegate = async (delegator_address, validator_src_address, validator_dst_address, amount) => {
+        const message = {
+            typeUrl: "/cosmos.staking.v1beta1.MsgBeginRedelegate",
+            value: {
+                delegatorAddress: delegator_address,
+                validatorSrcAddress: validator_src_address,
+                validatorDstAddress: validator_dst_address,
+                amount
+            }
+        }
+        return this.signAndBroadCast(delegator_address, [message]);
+    };
 
-    deposit = async (proposalId, depositor, amount, broadcastMode = broadcastModeObj.BROADCAST_MODE_BLOCK) => {
-        const wallet = await this.collectWallet();
-        const message = CosmosMessages.getMsgDepositProposal(proposalId, depositor, amount);
-        return this.broadcastMsg(wallet, await this.constructTxBody([message]), broadcastMode);
+    withdrawCommission = async (validator_address) => {
+        const message = {
+            type_url: "/cosmos.distribution.v1beta1.MsgWithdrawValidatorCommission",
+            value: {
+                validatorAddress: validator_address
+            }
+        }
+        return this.signAndBroadCast(validator_address, [message]);
+    };
+
+    createValidator = async (msg) => {
+        const message = {
+            type_url: '/cosmos.staking.v1beta1.MsgCreateValidator',
+            value: {
+                description: msg.description,
+                commission: msg.commission,
+                delegatorAddress: msg.delegator_address,
+                minSelfDelegation: msg.min_self_delegation,
+                pubkey: msg.pubkey,
+                validatorAddress: msg.validatorAddress,
+                value: msg.value,
+            }
+        }
+        return this.signAndBroadCast(msg.delegator_address, [message]);
     }
 
-    vote = async (proposalId, voter, option, broadcastMode = broadcastModeObj.BROADCAST_MODE_BLOCK) => {
-        const wallet = await this.collectWallet();
-        const message = CosmosMessages.getMsgVoteProposal(proposalId, voter, option);
-        return this.broadcastMsg(wallet, await this.constructTxBody([message]), broadcastMode);
+    deposit = async (proposalId, depositor, amount) => {
+        const message = {
+            type_url: '/cosmos.gov.v1beta1.MsgDeposit',
+            value: {
+                proposalId: Number(proposalId), depositor: depositor, amount
+            }
+        }
+        return this.signAndBroadCast(proposalId, [message]);
     }
 
-    withdrawDelegatorReward = async (msgs, broadcastMode = broadcastModeObj.BROADCAST_MODE_BLOCK) => {
-        const wallet = await this.collectWallet();
-        let messages = [];
-        for (let msg of msgs) {
-            messages.push(CosmosMessages.getMsgWithdrawDelegatorReward(msg.delegator_address, msg.validator_address));
+    vote = async (proposalId, voter, option) => {
+        const message = {
+            type_url: '/cosmos.gov.v1beta1.MsgDeposit',
```

Comment:
The type_url of vote should be MsgVote. I'll edit this for you

Reply:
Thank you for that